src/containers/announcements/ducks/thunks.ts

```diff
@@ -1,19 +1,21 @@
 import { Announcement, AnnouncementsThunkAction } from './types';
 import { announcements } from './actions';
 
-export const getAnnouncements = (limit?: number): AnnouncementsThunkAction<void> => {
-    return (dispatch, getState, { publicApiClient }) => {
-        dispatch(announcements.loading());
-        return publicApiClient
-            .getAnnouncements()
-            .then((response: Announcement[]) => {
-                return (limit && limit >= 0) ? response.slice(0, limit) : response;
-            })
-            .then((response: Announcement[]) => {
-                dispatch(announcements.loaded(response));
-            })
-            .catch((error: any) => {
-                dispatch(announcements.failed(error));
-            });
-    };
+export const getAnnouncements = (
+  limit?: number,
+): AnnouncementsThunkAction<void> => {
+  return (dispatch, getState, { publicApiClient }) => {
+    dispatch(announcements.loading());
+    return publicApiClient
+      .getAnnouncements()
+      .then((response: Announcement[]) => {
+        return limit && limit >= 0 ? response.slice(0, limit) : response;
```

Comment:
I think I made this comment earlier, but we should just store the entire response if we're already loading it, or we can change the API to return a subset. We would use a selector to slice based on the selected amount.